api/src/main/java/org/pmiops/workbench/cohortbuilder/CohortBuilderServiceImpl.java

```diff
@@ -275,33 +275,14 @@ public List<Criteria> findCriteriaAutoCompleteV2(
 
     SearchTerm searchTerm = new SearchTerm(term, mySQLStopWordsProvider.get().getStopWords());
 
-    List<DbCriteria> criteriaList = new ArrayList<>();
-
-    if (searchTerm.hasModifiedTermOnly()) {
-      criteriaList =
-          cbCriteriaDao.findCriteriaByDomainAndTypeAndStandardAndFullText(
-              domain, type, standard, searchTerm.getModifiedTerm(), pageRequest);
-    } else if (searchTerm.hasEndsWithOnly()) {
-      criteriaList =
-          cbCriteriaDao.findCriteriaByDomainAndTypeAndStandardAndNameEndsWith(
-              domain, type, standard, searchTerm.getEndsWithTerms(), pageRequest);
-    } else if (searchTerm.hasEndsWithTermsAndModifiedTerm()) {
-      criteriaList =
-          cbCriteriaDao.findCriteriaByDomainAndTypeAndStandardAndTermAndNameEndsWith(
-              domain,
-              type,
-              standard,
-              searchTerm.getModifiedTerm(),
-              searchTerm.getEndsWithTerms(),
-              pageRequest);
-    }
+    List<DbCriteria> criteriaList =
+        cbCriteriaDao.findCriteriaAutoComplete(domain, type, standard, searchTerm, pageRequest);
```

Comment:
Much cleaner service method with this logic in the dao.

Reply:
I like it!